### PR TITLE
Add API Key to Groups

### PR DESCRIPTION
**Pull request by georgedias**
This PR is to add the ability for associating API Key with groups. It is in support of issue number 3128 
(don't use the # as the issue would be deleted upon merging)

### CODE REVIEW

File: apps/backend/src/evaluations/evaluations.controller.ts [hunk 1]
```diff
@@ -102,7 +104,33 @@
...
       }
 
-      let groups: Group[] = createEvaluationDto.groups
+      // If the "user" is a group, we'll add the evaluation to the group, and ignore any other groups
+      if (request.user instanceof Group) {
+        const evaluation = await this.evaluationsService
+          .create({
+            // Only respect custom file names for single file uploads
+            filename:
+              data.length > 1
```
[the remark this answers is not in the document]
Comment (camdenmoors) [reply apps/backend/src/evaluations/evaluations.controller.ts:113]: This is always an array as opposed to `UploadedFile()`

File: apps/backend/src/evaluations/evaluations.controller.ts [hunk 2]
```diff
@@ -102,38 +104,70 @@
...
-        `${this.configService.get('EXTERNAL_URL') || ''}/results/${
-          evaluation.id
-        }`
-      );
-      return _.omit(createdDto, 'data');
+        const evaluation = await this.evaluationsService
+          .create({
+            // Only respect custom file names for single file uploads
+            filename:
+              data.length > 1
```
[the remark this answers is not in the document]
Comment (camdenmoors) [reply apps/backend/src/evaluations/evaluations.controller.ts:147]: This is always an array as opposed to `UploadedFile()`

**Comment by em-c-rod** [comment]
@camdenmoors  How does this impact the instructions for programmatically generating an API key (which was written for a user)? Could the same be done for the groups? Will the functionality for the user be the same?

**Comment by em-c-rod** [comment]
@camdenmoors Can you explain what you tested? There are no coded tests, so I want to make sure we are covering the different situations.

**Pull request merged**